Assets/BossRoom/Scripts/Server/Game/Character/PlayerServerCharacter.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections;
+using System.Collections.Generic;
+using MLAPI;
+using UnityEngine;
+
+namespace BossRoom.Server
+{
+    /// <summary>
+    /// Attached to the player-characters' prefab, this maintains a list of active ServerCharacter objects for players.
+    /// </summary>
+    /// <remarks>
+    /// This is an optimization. In server code you can already get a list of players' ServerCharacters by
+    /// iterating over the active connections and calling GetComponent() on their PlayerObject. But we need
+    /// to iterate over all players quite often -- the monsters' IdleAIState does so in every Update() --
+    /// and all those GetComponent() calls add up! So this optimization lets us iterate without calling
+    /// GetComponent().
+    /// </remarks>
+    [RequireComponent(typeof(ServerCharacter))]
+    public class PlayerServerCharacter : NetworkBehaviour
+    {
+        static List<ServerCharacter> s_ActivePlayers = new List<ServerCharacter>();
+
+        ServerCharacter m_CachedServerCharacter;
+
+        void OnEnable()
+        {
+            m_CachedServerCharacter = GetComponent<ServerCharacter>();
```

Comment:
Just last comment to serialize & cache this and we're good to go.

Reply:
done. I realized this class is also not properly disabling itself on client, so I changed that as well.